src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryFileValidator.cs

```diff
@@ -0,0 +1,153 @@
+                            result = false;
+                            break;
+                        }
+                    }
+                    else
+                    {
+                        // If the file path doesn't exist, check if the folder exists
+                        string folderPath = PathWrapper.GetDirectoryName(filePath);
+                        if (isLocal)
+                        {
+                            if (!string.IsNullOrEmpty(folderPath) && !Directory.Exists(folderPath))
+                            {
+                                errorMessage = string.Format(SR.InvalidBackupPathError, folderPath);
+                                result = false;
+                                break;
+                            }
+                        }
+                        else
+                        {
+                            bool isFolderOnRemote = true;
+                            bool existsOnRemote = IsPathExisting(serverConnection, folderPath, ref isFolderOnRemote);
+                            if (!existsOnRemote)
+                            {
+                                errorMessage = string.Format(SR.InvalidBackupPathError, folderPath);
+                                result = false;
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
+            else
+            {
+                result = false;
+            }
+
+            if (!string.IsNullOrEmpty(errorMessage))
+            {
+                throw new FileBrowserException(errorMessage);
+            }
+
+            return result;
+        }
+
+        #region private methods
+
+        private static bool IsPathExisting(ServerConnection serverConnection, string path, ref bool IsFolder)
+        {
+            Enumerator en = null;
+            DataSet ds = new DataSet();
+            ds.Locale = CultureInfo.InvariantCulture;
+            Request req = new Request();
+
+            en = new Enumerator();
+            req.Urn = "Server/File[@FullName='" + Urn.EscapeString(path) + "']";
+            ds = en.Process(serverConnection, req);
+            int iCount = ds.Tables[0].Rows.Count;
+
+            if (iCount > 0)
+            {
+                IsFolder = !(Convert.ToBoolean(ds.Tables[0].Rows[0]["IsFile"], CultureInfo.InvariantCulture));
+                return true;
+            }
+
+            IsFolder = false;
+            return false;
+        }
+
+        private static string GetMachineName(string sqlServerName)
+        {
+            string machineName = "";
+            if (sqlServerName != null)
+            {
+                // special case (local) which is accepted SQL(MDAC) but by OS
+                if ((sqlServerName.ToLowerInvariant().Trim() == localSqlServer) || (sqlServerName.ToLowerInvariant().Trim() == localMachineName))
```

Comment:
>sqlServerName [](start = 21, length = 13)

this will throw exception if the string doesn't have any '//'

Reply:
why? Trim() removes only whitespace.